web-library/src/main/java/ru/skypro/lessons/springboot/weblibrary/pojo/SecurityConfig.java

```diff
@@ -0,0 +1,81 @@
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AuthorizeHttpRequestsConfigurer;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.JdbcUserDetailsManager;
+import org.springframework.security.provisioning.UserDetailsManager;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+
+@EnableWebSecurity
+@Configuration
+public class SecurityConfig {
+
+    @Autowired
+    private UserDetailsService userDetailsService;
+    @Bean
+    public UserDetailsManager userDetailsManager(DataSource dataSource,
+                                                 AuthenticationManager authenticationManager) {
+
+        // Инициализируем JdbcUserDetailsManager с dataSource
+        // и authenticationManager для работы с базой данных
+        JdbcUserDetailsManager jdbcUserDetailsManager =
+                new JdbcUserDetailsManager(dataSource);
+
+        jdbcUserDetailsManager.setAuthenticationManager(authenticationManager);
+        return jdbcUserDetailsManager;
+    }
+
+    // Создаем бин authenticationManager и получаем его
+    // из AuthenticationConfiguration.
+    @Bean
+    public AuthenticationManager authenticationManager(AuthenticationConfiguration authenticationConfiguration)
+            throws Exception {
+        return authenticationConfiguration.getAuthenticationManager();
+    }
+
+    // Создаем бин SecurityFilterChain для настройки фильтра безопасности.
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http.csrf()
+                .disable()
+                .authorizeHttpRequests(this::customizeRequest);
+
+        return http.build();
+    }
+
+    // Метод для настройки прав доступа к URL на основе ролей пользователей
+    private void customizeRequest(AuthorizeHttpRequestsConfigurer<HttpSecurity>.AuthorizationManagerRequestMatcherRegistry registry) {
+        try {
+            registry.requestMatchers(new AntPathRequestMatcher("/admin/**"))
+                    .hasAnyRole("ADMIN")  // Только для пользователей с ролью ADMIN.
+                    .requestMatchers(new AntPathRequestMatcher("/**"))
+                    .hasAnyRole("USER")   // Только для пользователей с ролью USER.
+                    .and()
+                    .formLogin().permitAll()  // Разрешаем всем доступ к форме ввода.
+                    .and()
+                    .logout().logoutUrl("/logout");  // Устанавливаем URL
+            // для выхода из системы.
+
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
```

Comment:
Переписываем фильтрацию:

```suggestion
    @Bean
    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {

        http.csrf(csrf -> csrf.ignoringRequestMatchers("/*"))
                .authorizeHttpRequests(this::customizeRequest)
                .formLogin(Customizer.withDefaults())
                .logout(Customizer.withDefaults());
        return http.build();
    }

    private void customizeRequest(AuthorizeHttpRequestsConfigurer<HttpSecurity>.AuthorizationManagerRequestMatcherRegistry registry) {
        try {
            registry.requestMatchers(new AntPathRequestMatcher("/admin/**")).hasAnyRole(Role.ADMIN.name())
                    .requestMatchers(new AntPathRequestMatcher("/**")).hasAnyRole(Role.USER.name());

        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

```